modules/storage/manager.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package storage
+
+import (
+	"encoding/json"
+	"sort"
+	"sync"
+)
+
+var manager *Manager
+
+// Manager is the storage manager
+type Manager struct {
+	mutex sync.Mutex
+
+	storage map[string]ManagedStorage
+}
+
+// ManagedStorage represents an ObjectStorage coupled with its name, type and configuration
+type ManagedStorage struct {
+	Name          string
+	Type          string
+	Config        interface{}
+	ObjectStorage ObjectStorage
+}
+
+// ConfigAsString returns the config as a string
+func (ms *ManagedStorage) ConfigAsString() string {
+	bs, err := json.Marshal(ms.Config)
+	if err != nil {
+		return ""
+	}
+	return string(bs)
+}
+
+func init() {
+	_ = GetManager()
+}
+
+// GetManager returns a Manager and initializes one as singleton is there's none yet
+func GetManager() *Manager {
+	if manager == nil {
+		manager = &Manager{
+			storage: make(map[string]ManagedStorage),
+		}
+	}
+	return manager
+}
+
+// Get will return the storage for the provided name
+func (m *Manager) Get(name string) ObjectStorage {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	return m.storage[name].ObjectStorage
```

Comment:
what happen if there is no storage for name ...
more specific: can do a ObjectStorage with no values do harmfull things?